src/Common/Commands.Common.Authentication/Authentication/ManagedServiceTokenInfo.cs

```diff
@@ -42,9 +44,22 @@ public class ManagedServiceTokenInfo
         [JsonProperty(PropertyName = "token_type")]
         public string TokenType { get; set; }
 
+        [JsonIgnore]
+        public DateTime CreationDate { get; } = DateTime.UtcNow;
+
         public override string ToString()
         {
             return $"(AccessToken: {AccessToken}, ExpiresIn: {ExpiresIn}, ExpiresOn: {ExpiresOn}, Resource:{Resource})";
         }
+
+        public bool ShouldCache()
+        {
+            return !IsExpired();
+        }
+
+        public bool IsExpired()
+        {
+            return DateTime.UtcNow - CreationDate > (TimeSpan.FromSeconds(ExpiresIn) - TimeSpan.FromMinutes(5));
```

Comment:
Is the 5 minutes supposed to be a buffer time? So that we never end up with an expired token in our cache? Also is this related to the check for 4 mins in the code above?

If so, should we look at introducing a constant that is set to 5 and use <said constant> -1 in the code above?